arch/risc-v/include/bl602/irq.h

```diff
@@ -224,8 +224,49 @@ extern "C"
  * Public Function Prototypes
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: up_irq_enable
+ *
+ * Description:
+ *   Return the current interrupt state and enable interrupts
+ *
+ ****************************************************************************/
+
 EXTERN irqstate_t up_irq_enable(void);
 
+/****************************************************************************
+ * Name: up_irq_save
+ *
+ * Description:
+ *   Save the current interrupt state and disable interrupts.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   Interrupt state prior to disabling interrupts.
+ *
+ ****************************************************************************/
+
+EXTERN irqstate_t up_irq_save(void);
```

Comment:
This seems a bit weird.
`up_irq_save` is already declared at `arch/risc-v/include/irq.h`, which is already included in `bl602_irq.c` as `<arch/irq.h>`.
If this patch is really necessary, there might be something wrong with the include paths.

Reply:
@gustavonihei - it may very well be that the added lines in arch/risc-v/include/bl602/irq.h  are not needed but the inclusion into arch.h is conflicting with the arch specific deceleration.